cosmos/errors.go

```diff
@@ -0,0 +1,26 @@
+package cosmos
+
+import (
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// CodespaceMesg is a cosmos codespace for all mesg errors.
+	CodespaceMesg types.CodespaceType = "mesg"
+)
+
+const (
+	// Base mesg codes.
+	CodeMesgInternal   types.CodeType = 1000
+	CodeMesgValidation                = 2000
```

Comment:
Why 1000 and then 2000? Why not 1001?
Also the variable name can simply be `CodeValidation`
```suggestion
	CodeValidation                = 1001
```

Reply:
As we discussed, they are different types of error.

1) 1st 1000 is reserved for errors
2) 2nd 1000 is reserverd for validation error - in the future we might want have different validation errors like: HashValidation, DataValidation, AccountValidation, etc

Reply:
oh ok. do we want to have so specific error code (HashValidation, DataValidation, etc..)? I would go with just one CodeValidation for now and a nice error message

Reply:
Maybe we will want. I started to numerate such errors from 2000 exactly for that purpose. It the future we migth have password, hash, data, user, accout, and more validations. And by setting it to 1001 we don't have such option, and what are the pros?